astronomer/providers/amazon/aws/triggers/emr.py

```diff
@@ -80,60 +85,27 @@ async def run(self) -> AsyncIterator["TriggerEvent"]:  # type: ignore[override]
             yield TriggerEvent({"status": "error", "message": str(e)})
 
 
-class EmrContainerOperatorTrigger(BaseTrigger):
-    """
-    The EmrContainerSensorTrigger is triggered when EMR container is created, it polls for the AWS EMR EKS Virtual
-    Cluster Job status. It is fired as deferred class with params to run the task in trigger worker
-
-    :param virtual_cluster_id: The EMR on EKS virtual cluster ID.
-    :param name: The name of the job run.
-    :param execution_role_arn: The IAM role ARN associated with the job run.
-    :param release_label: The Amazon EMR release version to use for the job run.
-    :param job_driver: Job configuration details, e.g. the Spark job parameters.
-    :param configuration_overrides: The configuration overrides for the job run,
-            specifically either application configuration or monitoring configuration.
-    :param client_request_token: The client idempotency token of the job run request.
-    :param aws_conn_id: Reference to AWS connection id.
-    :param poll_interval: polling period in seconds to check for the status.
-    :param max_retries: maximum retry for poll for the status.
-    """
-
-    INTERMEDIATE_STATES: List[str] = ["PENDING", "SUBMITTED", "RUNNING"]
-    FAILURE_STATES: List[str] = ["FAILED", "CANCELLED", "CANCEL_PENDING"]
-    SUCCESS_STATES: List[str] = ["COMPLETED"]
-    TERMINAL_STATES: List[str] = ["COMPLETED", "FAILED", "CANCELLED", "CANCEL_PENDING"]
-
-    def __init__(
-        self,
-        virtual_cluster_id: str,
-        name: str,
-        job_id: str,
-        aws_conn_id: str = "aws_default",
-        poll_interval: int = 30,
-        max_tries: Optional[int] = None,
-    ):
-        super().__init__()
-        self.virtual_cluster_id = virtual_cluster_id
-        self.name = name
-        self.job_id = job_id
-        self.aws_conn_id = aws_conn_id
-        self.poll_interval = poll_interval
-        self.max_tries = max_tries
+class EmrContainerOperatorTrigger(EmrContainerBaseTrigger):
+    """Poll for the status of EMR container until reaches terminal state"""
 
     def serialize(self) -> Tuple[str, Dict[str, Any]]:
         """Serializes EmrContainerOperatorTrigger arguments and classpath."""
         return (
             "astronomer.providers.amazon.aws.triggers.emr.EmrContainerOperatorTrigger",
             {
                 "virtual_cluster_id": self.virtual_cluster_id,
-                "name": self.name,
                 "job_id": self.job_id,
                 "aws_conn_id": self.aws_conn_id,
                 "max_tries": self.max_tries,
                 "poll_interval": self.poll_interval,
             },
         )
 
+    INTERMEDIATE_STATES: List[str] = ["PENDING", "SUBMITTED", "RUNNING"]
+    FAILURE_STATES: List[str] = ["FAILED", "CANCELLED", "CANCEL_PENDING"]
+    SUCCESS_STATES: List[str] = ["COMPLETED"]
+    TERMINAL_STATES: List[str] = ["COMPLETED", "FAILED", "CANCELLED", "CANCEL_PENDING"]
+
```

Comment:
Would be good if we move to this before the first method of the class at the top for readability since these are class variables.

Reply:
yeah, somehow I missed it. Fixed here cdea7279867152df20182404e5602934a2cb3c61